src/main/java/edu/tamu/app/service/ProjectFactory.java

```diff
@@ -427,4 +427,24 @@ public void startProjectFileListeners() {
         });
     }
 
+    /**
+     * Initiate project file listener for single project.
+     *
+     * @param id
+     *   ID of the project to initiate listener for.
+     */
+    public void startProjectFileListener(Long id) {
```

Comment:
Can you refactor this so that startProjectFileListeners() and startProjectFileListener() share a common internal method?

`    public void startProjectFileListeners() {`
`        projectRepo.findAll().forEach(project -> {`
`            activateProjectFileListener(project);`
`        });`
`    }`

    /**
     * Initiate project file listener for single project.
     *
     * @param id
     *   ID of the project to initiate listener for.
     */
    public void startProjectFileListener(Long id) {
        activateProjectFileListener(projectRepo.findOne(id));
    }`